### PR TITLE
add parse_datevalue()

### PR DESCRIPTION
**Pull request by smn**
Allows us to do `parse_datevalue("2016-02-29T22:25:00-00:00", "%FT%T%:z")` and get back `~U[2016-02-29 22:25:00Z]
`

### CODE REVIEW

File: lib/expression/callbacks/standard.ex [hunk 1]
```diff
@@ -145,6 +145,28 @@ defmodule Expression.Callbacks.Standard do
...
   end
 
+  @doc """
+  Parse random dates and times with `strftime` patterns and return a DateTime value
+  when it matches.
+  """
+  @expression_doc doc: "Parse a date value using strftime formatting and return a DateTime",
+                  expression: "parse_datevalue(\"2016-02-29T22:25:00-00:00\", \"%FT%T%:z\")",
+                  result: DateTime.new!(~D[2016-02-29], ~T[22:25:00])
+  @expression_doc doc: "Attempt to parse a date value and return nil when failing",
```
Comment (fedme) [inline lib/expression/callbacks/standard.ex:155]: are these doctested? If not could we add tests?

**Review by fedme — APPROVED** [review]
lgtm

File: lib/expression/callbacks/standard.ex (hunk 1 above)
Comment (smn) [reply lib/expression/callbacks/standard.ex:155]: these @expression_docs are compiled by the macro to become doctests and documentation at the same time

Comment (fedme) [reply lib/expression/callbacks/standard.ex:155]: really impressive, so cool!

Comment (smn) [reply lib/expression/callbacks/standard.ex:155]: https://hexdocs.pm/expression/Expression.Callbacks.Standard.html#parse_datevalue/3 are the docs + doctests

**Pull request merged**